src/versionbits.cpp

```diff
@@ -49,20 +56,17 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
         pindexPrev = vToCompute.back();
         vToCompute.pop_back();
 
+        // We track state by previous-block, so the height we should be comparing is +1
+        const int64_t height = pindexPrev->nHeight + 1;
+
         switch (state) {
             case ThresholdState::DEFINED: {
-                if (pindexPrev->GetMedianTimePast() >= nTimeTimeout) {
-                    stateNext = ThresholdState::FAILED;
```

Comment:
I think that's being checked by the test cases.

Reply:
The fuzz test has all transitions of the state machine implemented (in a "come from" form rather than a "go to" one). I don't think it's possible to make incorrect transitions that it won't detect.